### PR TITLE
fix(output): fixes handling of untrusted URLs in output/url

### LINKED ISSUE #9146: Only set rel="nofollow" for external links
Any reason why we don't trust internal links by default? I propose we only default `is_trusted` flag to `false` if the link is outside an Elgg installation. 

See https://github.com/Elgg/Elgg/issues/9117

### PR DESCRIPTION
**Pull request by hypeJunction**
All untrusted URLs are now stripped of tags. Rel is set to "nofollow" only on external links.

Fixes #9146

### CODE REVIEW

File: views/default/output/url.php [hunk 1]
```diff
@@ -57,9 +57,9 @@
 	}
 
 	if (!elgg_extract('is_trusted', $vars, false)) {
-		if (!isset($vars['rel'])) {
+		$url = strip_tags($url);
+		if (!isset($vars['rel']) && parse_url($url, PHP_URL_HOST) != parse_url(elgg_get_site_url(), PHP_URL_HOST)) {
```
Comment (mrclay) [inline views/default/output/url.php]: Use a variable or two to clean up?

**Comment by hypeJunction** [comment]
@mrclay Added 3!

File: views/default/output/url.php [hunk 2]
```diff
@@ -57,9 +57,15 @@
 	}
 
 	if (!elgg_extract('is_trusted', $vars, false)) {
+		$url = strip_tags($url);
 		if (!isset($vars['rel'])) {
-			$vars['rel'] = 'nofollow';
-			$url = strip_tags($url);
+			$url_host = parse_url($url, PHP_URL_HOST);
+			$site_url = elgg_get_site_url();
+			$site_url_host = pase_url($site_url, PHP_URL_HOST);
```
Comment (mrclay) [inline views/default/output/url.php]: pase_url :)

Comment (hypeJunction) [inline views/default/output/url.php]: F******

**Comment by juho-jaakkola** [comment]
LGTM ( but haven't tested, so not 100% sure whether `pase_url` works)

**Comment by hypeJunction** [comment]
@juho-jaakkola :facepunch:

**Comment by hypeJunction** [comment]
@mrclay @juho-jaakkola URLs are not longer pased

**Comment by hypeJunction** [comment]
Can our tests require each view file to make sure typos like that don't end up in core?

**Comment by mrclay** [comment]
Ideally, sure. Many would probably have to be done in the Simpletests due to calling out for resources.

File: views/default/output/url.php [hunk 3]
```diff
@@ -57,9 +57,15 @@
 	}
 
 	if (!elgg_extract('is_trusted', $vars, false)) {
```
Comment (mrclay) [inline views/default/output/url.php]: Hmm, logic isn't quite right. If `$vars['is_trusted']` was set to `false` we want to take it on its word, right?

**Comment by hypeJunction** [comment]
I don't mean using `elgg_view()`. Just iterated and `include` each view file once. That will throw Fatal Error if something is pasing rather than parsing.

File: views/default/output/url.php (hunk 3 above)
Comment (mrclay) [inline views/default/output/url.php]:
```php
$is_trusted = elgg_extract('is_trusted', $vars);
if ($is_trusted === null) {
    // compare hosts
}
```

Comment (hypeJunction) [inline views/default/output/url.php]: I suppose.

**Comment by mrclay** [comment]
It will also throw fatals in lots of correct views because the PHPUnit environment isn't a complete install.

File: views/default/output/url.php [hunk 4]
```diff
@@ -56,10 +56,20 @@
...
 	}
 
-	if (!elgg_extract('is_trusted', $vars, false)) {
+	$is_trusted = elgg_extract('is_trusted', $vars);
+	if (!$is_trusted) {
+		$url = strip_tags($url);
 		if (!isset($vars['rel'])) {
-			$vars['rel'] = 'nofollow';
-			$url = strip_tags($url);
+			if ($is_trusted === null) {
```
Comment (hypeJunction) [inline views/default/output/url.php]: @mrclay Better?

**Comment by mrclay** [comment]
New logic looks good.

File: views/default/output/url.php [hunk 5]
```diff
@@ -56,10 +56,20 @@
 		$url = elgg_add_action_tokens_to_url($url, false);
 	}
 
-	if (!elgg_extract('is_trusted', $vars, false)) {
+	$is_trusted = elgg_extract('is_trusted', $vars);
+	if (!$is_trusted) {
```
Comment (hypeJunction) [inline views/default/output/url.php]: I will pretend `0` doesn't exist :baby_chick:

**Pull request merged**